chromium_src/chrome/browser/profiles/delete_profile_helper.cc

```diff
@@ -32,7 +32,7 @@ bool StopSyncIfActive(Profile* profile, const char* func_name) {
     syncer::BraveSyncServiceImpl* sync_service =
         static_cast<syncer::BraveSyncServiceImpl*>(
             SyncServiceFactory::GetForProfile(profile));
-    sync_service->StopAndClear();
+    sync_service->StopAndClearWithShutdownReason();
```

Comment:
what does upstream do here? I'm a bit confused as to why we need to expose these methods when they do not

Reply:
cc: @cdesouza-chromium

Reply:
cc @AlexeyBarabash as is based on some new underlying code, and it looks like upstream was already not making use of this call when it was introduced in brave. The CL[1] that introduced this uses a different way to trigger stop and clear on tests, but I'm not sure that would be appropriate in production code.

Notice that we are now having to override `SyncServiceImpl::StopAndClear`, as `ServiceImpl::StopAndClear` was not only being called in the public interface, but also overridden as a customisation point. I think Alexey may be better equipped to answer/advise on this issue.

[1] https://crrev.com/c/5800573

Reply:
This isn't blocking, but I'm concerned that we're doing something different from upstream here and it's not clear why

Reply:
@bridiver

All usages of former `SyncServiceImpl::StopAndClear` are at `brave_sync_service_impl.cc`, we follow the upsteeam here.

The only place where it doesn't is `delete_profile_helper.cc`. We need to stop sync here to avoid DELETE records propagation to other devices in the chain.
Related PR: https://github.com/brave/brave-core/pull/24988 .

Upstream doesn't need to stop sync service explicitly when deleting profile because they do call 
```
identity_manager->GetPrimaryAccountMutator()->ClearPrimaryAccount(
      signin_metrics::ProfileSignout::kSignoutDuringProfileDeletion);
```
[link](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/profiles/delete_profile_helper.cc;l=90;drc=dabf53d870a39f6ea7b4e5af6368d54a7e8974eb;bpv=0;bpt=0?q=chrome%2Fbrowser%2Fprofiles%2Fdelete_profile_helper.cc&ss=chromium%2Fchromium%2Fsrc)

For Brave [PR description](https://github.com/brave/brave-core/pull/24988#issue-2448661510)

> Upstream's code doesn't stops Brave sync service, because identity_manager->HasPrimaryAccount gives false. And even more, the next line identity_manager->GetPrimaryAccountMutator()->ClearPrimaryAccount() doesn't actually cause stop sending delete record